relayer/client-tx.go

```diff
@@ -111,3 +135,36 @@ func (c *Chain) UpdateClients(dst *Chain) (err error) {
 
 	return nil
 }
+
+// UpgradesClients upgrades the client on src after dst chain has undergone an upgrade.
+func (c *Chain) UpgradeClients(dst *Chain) error {
+	sh, err := NewSyncHeaders(c, dst)
+	if err != nil {
+		return err
+	}
+	sh.Updates(c, dst)
+	height := int64(sh.GetHeight(dst.ChainID))
```

Comment:
I believe this is right
```suggestion
	height := int64(sh.GetHeight(dst.ChainID)) -1
```

but we can test live